base/applications/rapps/available.cpp

```diff
@@ -83,6 +83,19 @@ VOID CAvailableApplicationInfo::RetrieveGeneralInfo(AvailableStrings& AvlbString
         }
     }
 
+    // TODO: are we going to support specify an URL for an icon ?
+    ATL::CStringW IconLocation;
+    if (GetString(L"Icon", IconLocation))
+    {
+        // TODO: Does the filename contain anything stuff like "\\" ".." ":" "<" ">" ?
```

Comment:
`wcspbrk`, with `L"\\/:*?\"<>|"` will probably get you started :)

Reply:
I think "\\" and ".." should be acceptable as long as it's not used for dir escape (as we might want to classify those icons/screenshots into different folders). Now I've already done some work about this.

and about stuff like ```<>?*|:```, I think it over and thought they won't cause some actual damage. the worst thing could be is the file can not be opened. (If I'm wrong and too optimistic, please tell me). So I decided to just leave it there. 🤣